query/src/catalogs/impls/catalog/overlaid_catalog.rs

```diff
@@ -190,6 +190,19 @@ impl Catalog for OverlaidCatalog {
         &self,
         plan: CreateDatabasePlan,
     ) -> common_exception::Result<CreateDatabaseReply> {
+        match self.read_only.get_database(&plan.db).await {
+            Ok(_) => {
+                return Err(ErrorCode::DatabaseAlreadyExists(format!(
+                    "{} database exists",
+                    plan.db
+                )));
+            }
+            Err(err) => {
+                if err.code() != ErrorCode::UnknownDatabase("").code() {
```

Comment:
Why we check the `UnknownDatabase` error code here not return the error?

Reply:
It seems we need a `self.read_only.exists_database` returns bool or true api here

Reply:
when creating a DB, if the DB does not exist in read_only, it should be able to create in `bottom` and other errors should be return.

Reply:
should I add the func `exists_database` in trait `Catalog`?

Reply:
Make sense.